opentelemetry/proto/resource/v1/resource.proto

```diff
@@ -34,4 +34,39 @@ message Resource {
   // dropped_attributes_count is the number of dropped attributes. If the value is 0, then
   // no attributes were dropped.
   uint32 dropped_attributes_count = 2;
+
+  // Set of entities that participate in this Resource.
+  // Status: [Development]
+  repeated ResourceEntityRef entity_refs = 3;
 }
+
+// Status: [Development]
+message ResourceEntityRef {
+  // The Schema URL, if known. This is the identifier of the Schema that the entity data
+  // is recorded in. To learn more about Schema URL see
+  // https://opentelemetry.io/docs/specs/otel/schemas/#schema-url
+  //
+  // This schema_url applies to the data in this message and to the Resource attributes
+  // referenced by id_attr_keys and descr_attr_keys.
+  // TODO: discuss if we are happy with this somewhat complicated definition of what
+  // the schema_url applies to.
+  //
+  // This field obsoletes the schema_url field in ResourceMetrics/ResourceSpans/ResourceLogs.
+  string schema_url = 1;
+
+  // Defines the type of the entity. MUST not change during the lifetime of the entity.
+  // For example: "service" or "host". This field is required and MUST not be empty
+  // for valid entities.
+  string type = 2;
+
+  // Attribute Keys that identify the entity.
+  // MUST not change during the lifetime of the entity. The Id must contain at least one attribute.
+  // These keys MUST exist in the containing Resource.attributes.
+  repeated string id_attr_keys = 3;
```

Comment:
Adding this, but don't want to start a huge bikeshedding argument.

Let's be confident we are ok with these names in json.

Alternative suggestions (in order of my preference):

- `id_keys`, `desc_keys`
- `id_keys`, `descr_keys`
- `identifying_keys`, `descriptive_keys`
- `id_attr_keys`, `desc_attr_keys`
- `id_refs`, `desc_refs`

Reply:
TBH, I still do not think that having a schema and a set of keys is a good idea; I would expect that having a schema only and baking both key sets into the code is a far better solution; this is data that we would ship over the network all the time